tests/models/test_taskinstance.py

```diff
@@ -1670,6 +1754,100 @@ def test_get_rendered_template_fields(self):
         with create_session() as session:
             session.query(RenderedTaskInstanceFields).delete()
 
+    def validate_ti_states(self, dag_run, ti_state_mapping):
+        for task_id, expected_state in ti_state_mapping.items():
+            task_instance = dag_run.get_task_instance(task_id=task_id)
+            self.assertEqual(task_instance.state, expected_state)
+
+    @parameterized.expand([
+        (
+            {('scheduler', 'schedule_after_task_execution'): 'True'},
+            {'A': 'B', 'B': 'C'},
+            {'A': State.QUEUED, 'B': State.NONE, 'C': State.NONE},
+            {'A': State.SUCCESS, 'B': State.SCHEDULED, 'C': State.NONE},
+            {'A': State.SUCCESS, 'B': State.SUCCESS, 'C': State.SCHEDULED}
+        ),
+        (
+            {('scheduler', 'schedule_after_task_execution'): 'False'},
+            {'A': 'B', 'B': 'C'},
+            {'A': State.QUEUED, 'B': State.NONE, 'C': State.NONE},
+            {'A': State.SUCCESS, 'B': State.NONE, 'C': State.NONE},
+            None
+        ),
+        (
+            {('scheduler', 'schedule_after_task_execution'): 'True'},
+            {'A': 'B', 'C': 'B', 'D': 'C'},
+            {'A': State.QUEUED, 'B': State.NONE, 'C': State.NONE, 'D': State.NONE},
+            {'A': State.SUCCESS, 'B': State.NONE, 'C': State.NONE, 'D': State.NONE},
+            None
```

Comment:
Could you add a comment here saying what we're testing, why we expect B to not be scheduled etc -- this will help Future Us